lib/livebook_web/live/hub_live/fly_component.ex

```diff
@@ -0,0 +1,197 @@
+                >
+                  <div class="rounded h-5 w-5" style={"background-color: #{@data["color"]}"}></div>
+                </div>
+                <div class="relative grow">
+                  <%= text_input(f, :color,
+                    value: @data["color"],
+                    class: "input",
+                    spellcheck: "false",
+                    required: true,
+                    maxlength: 7
+                  ) %>
+                  <button
+                    class="icon-button absolute right-2 top-1"
+                    type="button"
+                    phx-click="randomize_color"
+                    phx-target={@myself}
+                  >
+                    <.remix_icon icon="refresh-line" class="text-xl" />
+                  </button>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          <%= submit("Save", class: "button-base button-blue", phx_disable_with: "Saving...") %>
+        <% end %>
+      </.form>
+    </div>
+    """
+  end
+
+  defp load_data(%{assigns: %{operation: :new}} = socket) do
+    assign(socket, data: %{}, organizations: [], orgs_data: [])
+  end
+
+  defp load_data(%{assigns: %{operation: :edit, provider_id: id}} = socket) do
+    fly = Hubs.fetch_fly!(id)
+    organizations = [fly.organization]
+
+    data =
+      fly
+      |> Map.from_struct()
+      |> Map.delete(:organization)
+      |> Jason.encode!()
+      |> Jason.decode!()
+
+    opts = select_options(organizations, fly.id)
+
+    assign(socket, data: data, organizations: opts, orgs_data: organizations)
+  end
+
+  @impl true
+  def handle_event("fetch_data", %{"fly" => %{"token" => token}}, socket) do
+    case Fly.fetch_organizations(token) do
+      {:ok, organizations} ->
+        data = %{"token" => token, "color" => User.random_hex_color()}
+        opts = select_options(organizations)
+
+        {:noreply, assign(socket, data: data, organizations: opts, orgs_data: organizations)}
+
+      {:error, _} ->
+        send(self(), {:flash_error, "Invalid Access Token"})
+        {:noreply, assign(socket, data: %{}, organizations: [], orgs_data: [])}
+    end
+  end
+
+  def handle_event("randomize_color", _, socket) do
+    data = put_in(socket.assigns.data, ["color"], User.random_hex_color())
+
+    {:noreply, assign(socket, data: data)}
+  end
+
+  def handle_event("save_hub", %{"fly" => params}, socket) do
+    case socket.assigns.operation do
+      :new -> create_fly(socket, params)
+      :edit -> update_fly(socket, params)
+    end
+  end
+
+  def handle_event("update_data", %{"fly" => data}, socket) do
+    opts = select_options(socket.assigns.orgs_data, data["id"])
+
+    {:noreply, assign(socket, data: data, organizations: opts)}
+  end
+
+  defp select_options(orgs, org_id \\ nil) do
+    for org <- orgs do
+      opts = if org.id == org_id, do: [selected: true], else: []
+
+      [key: org.name, value: org.id] ++ opts
```

Comment:
would this work?

```suggestion
      [key: org.name, value: org.id, selected: org.id == org_id]
```

in other words, it won't hurt to set `selected: false`, would it?

Reply:
yeah, it works much better